### PR TITLE
Extract migration for #582

### PR DESCRIPTION
**Pull request by okkez**
We should merge this PR before merge #582

### CODE REVIEW

File: db/migrate/20160719005841_create_categorizations.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class CreateCategorizations < ActiveRecord::Migration
+  def up
+    create_table :categorizations do |t|
+      t.references :category, foreign_key: true, null: false
+      t.references :post, foreign_key: true, null: false
+      t.integer :order, null: false
+
+      t.index %i(post_id category_id), unique: true
```
Comment (tricknotes) [inline db/migrate/20160719005841_create_categorizations.rb:8]: Maybe unique index for `order` is needed?

``` ruby
t.index %i(post_id order), unique: true
```

Comment (okkez) [inline db/migrate/20160719005841_create_categorizations.rb:8]: Added and force pushed changes.

File: db/structure.sql [hunk 2]
```diff
@@ -1126,9 +1205,20 @@ INSERT INTO schema_migrations (version) VALUES ('20160712070026');
...
 
+INSERT INTO schema_migrations (version) VALUES ('20160802025651');
+
+<<<<<<< HEAD
+=======
+INSERT INTO schema_migrations (version) VALUES ('20160804072008');
+
+INSERT INTO schema_migrations (version) VALUES ('20160804090315');
+
+>>>>>>> 26c972a... Add feature flag for hierarchical categories
```
Comment (mtsmfm) [inline db/structure.sql]: Conflict marker...

Comment (tricknotes) [inline db/structure.sql]: ouch!

**Pull request merged**